pkg/cluster_agent/resource_sender.py

```diff
@@ -0,0 +1,90 @@
+"""
+K8s Resource sender
+"""
+import json
+import logging
+from typing import Dict
+from datetime import datetime
+from traceback import format_exc
+from requests import post, exceptions as requests_exceptions
+from requests.auth import HTTPBasicAuth
+from encoders import DateTimeEncoder
+from cluster_scanner import ResourceScanResult
+
+def _resource_scan_result_to_dict(
+        scan_result: ResourceScanResult,
+        cluster_name: str,
+        update_time: datetime
+) -> Dict:
+    """
+    Parses resource scan result and formats it to a dict to be sent to Epsagon
+    """
+    data = {
+        "update_time": update_time.timestamp(),
+        "cluster": {
+            "name": cluster_name,
+        }
+    }
+    if scan_result.cluster_version:
```

Comment:
I think this can no longer happen, as you check for cluster_version as indication that the scan succeeded.
(But if you change the scanner behavior to throw error on failure then this is great :))

Reply:
right : )